File: 1480-Running Sum of 1d Array/golang/solution.go
package solution

func runningSum(nums []int) []int {
	var sum int = 0
	solve := make([]int, len(nums))

	for idx, val := range nums {
		sum += val
		solve[idx] = sum
	}
	return solve
}

// 4 ms, faster than 40.00% of Go online submissions for Running Sum of 1d Array.
// 2.8 MB, less than 100.00% of Go online submissions for Running Sum of 1d Array.

File: 1480-Running Sum of 1d Array/c++/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> solve;
        int sum = 0;
        
        for(int idx = 0; idx < nums.size(); ++idx) {
            sum += nums[idx];
            solve.push_back(sum);
        }
        return solve;
    }
};

// 0 ms, faster than 100.00% of C++ online submissions for Running Sum of 1d Array.
// 8.7 MB, less than 40.00% of C++ online submissions for Running Sum of 1d Array.